ksuid/ksuid.py

```diff
@@ -35,34 +23,44 @@ class ByteArrayLengthException(Exception):
 class Ksuid:
     """Ksuid class inspired by https://github.com/segmentio/ksuid"""
 
-    __uid: bytes
+    # Timestamp is a uint32
+    TIMESTAMP_LENGTH_IN_BYTES = 4
+
+    # Payload is 16-bytes
+    PAYLOAD_LENGTH_IN_BYTES = 16
+
+    # The length in bytes
+    BYTES_LENGTH = TIMESTAMP_LENGTH_IN_BYTES + PAYLOAD_LENGTH_IN_BYTES
+
+    # The length of the base64 representation (str)
+    BASE62_LENGTH = math.ceil(BYTES_LENGTH * 4 / 3)
+
+    _uid: bytes
 
     @classmethod
     def from_base62(cls: t.Type[SelfT], data: str) -> SelfT:
         """initializes Ksuid from base62 encoding"""
 
-        return cls.from_bytes(int.to_bytes(int(base62.decode(data)), BYTES_LENGTH, "big"))
+        return cls.from_bytes(int.to_bytes(int(base62.decode(data)), cls.BYTES_LENGTH, "big"))
 
     @classmethod
     def from_bytes(cls: t.Type[SelfT], value: bytes) -> SelfT:
         """initializes Ksuid from bytes"""
 
-        if len(value) != TIMESTAMP_LENGTH_IN_BYTES + PAYLOAD_LENGTH_IN_BYTES:
+        if len(value) != cls.TIMESTAMP_LENGTH_IN_BYTES + cls.PAYLOAD_LENGTH_IN_BYTES:
             raise ByteArrayLengthException()
 
         res = cls()
-        res.__uid = value
+        res._uid = value
 
         return res
 
-    def __init__(self, datetime: t.Optional[datetime] = None, payload: t.Optional[bytes] = None):
-        if payload is not None and len(payload) != PAYLOAD_LENGTH_IN_BYTES:
+    def __init__(self, dt: t.Optional[datetime] = None, payload: t.Optional[bytes] = None):
+        if payload is not None and len(payload) != self.PAYLOAD_LENGTH_IN_BYTES:
```

Comment:
I don't love breaking the API here, perhaps it would be better to import datetime with an alias instead?

Reply:
I agree, I was just being lazy about doing the right fix. Gimme a  moment.